pkg/collector/corechecks/snmp/config_validate_enrich.go

```diff
@@ -0,0 +1,80 @@
+package snmp
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func validateEnrichMetricTags(metricTags []metricTagConfig) []string {
+	var errors []string
+	for i := range metricTags {
+		errors = append(errors, validateEnrichMetricTag(&metricTags[i], metricsConfig{})...)
+	}
+	return errors
+}
+
+func validateEnrichMetrics(metrics []metricsConfig) []string {
```

Comment:
> Enrich makes me think there's some data added to the parsed yaml

There is actually one case of enrichment, we are validating the regex and storing the compiled regex:

https://github.com/DataDog/datadog-agent/blob/4efbd8fa26b871bab4da8cc7fb501a76873de743/pkg/collector/corechecks/snmp/config_validate_enrich.go#L64-L69

The rational for using one single validateEnrichMetrics method instead of two separate methods (validate/enrich) are to avoids some code duplication for traversing the `[]metricsConfig` and regex validation/compiling is both validation and enrichment.

Reply:
This behaviour could be more explicit, maybe in a comment before the function declaration

Reply:
Sure, added a function doc here: https://github.com/DataDog/datadog-agent/pull/6749/commits/923b1353803bb4311188c933382ef14f736202ce